src/main/java/spoon/support/compiler/jdt/JDTTreeBuilder.java

```diff
@@ -775,6 +802,9 @@ private boolean isCorrectTypeReference(TypeReference ref) {
 					? factory.Package().getOrCreate(concatSubArray(namesParameterized, index)).getReference()
 					: factory.Package().topLevel();
 			inner.setPackage(packageReference);
+
+			setPositionReference(res, ref);

```

Comment:
Why do you have this invocation here and [there](https://github.com/INRIA/spoon/pull/718/files#diff-072d4b8ceab9c4e090f675ba9342b6f9R779)?